<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>variable_from_to_scale</class>
 <widget class="QWidget" name="variable_from_to_scale">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>510</width>
    <height>80</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="MinimumExpanding">
    <horstretch>1</horstretch>
    <verstretch>1</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>80</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>16777215</width>
    <height>100</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <item>
    <widget class="QFrame" name="variable_from_to_frame">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="MinimumExpanding">
       <horstretch>1</horstretch>
       <verstretch>5</verstretch>
      </sizepolicy>
     </property>
     <property name="frameShape">
      <enum>QFrame::Shape::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Shadow::Raised</enum>
     </property>
     <layout class="QGridLayout" name="gridLayout_4" rowstretch="1,0" columnstretch="1,0,0,0">
      <property name="sizeConstraint">
       <enum>QLayout::SizeConstraint::SetMaximumSize</enum>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <property name="verticalSpacing">
       <number>0</number>
      </property>
      <item row="0" column="1">
       <widget class="QLabel" name="From_l">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
          <strikeout>false</strikeout>
          <kerning>true</kerning>
         </font>
        </property>
        <property name="text">
         <string>from</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="Var_l">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
          <strikeout>false</strikeout>
          <kerning>true</kerning>
         </font>
        </property>
        <property name="text">
         <string>Variable</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="floatLineEdit" name="to_entry">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
          <horstretch>1</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>1.00</string>
        </property>
        <property name="minValue" stdset="0">
         <double>0.000000000000000</double>
        </property>
        <property name="maxValue" stdset="0">
         <double>1000000000.000000000000000</double>
        </property>
        <property name="decimals" stdset="0">
         <number>10</number>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QComboBox" name="Var_dd">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
          <horstretch>1</horstretch>
          <verstretch>1</verstretch>
         </sizepolicy>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
          <strikeout>false</strikeout>
          <kerning>true</kerning>
         </font>
        </property>
        <item>
         <property name="text">
          <string>Param 2</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Param 1</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QLabel" name="to_l">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
          <strikeout>false</strikeout>
          <kerning>true</kerning>
         </font>
        </property>
        <property name="text">
         <string>to</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="floatLineEdit" name="from_entry">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
          <horstretch>1</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>0.00</string>
        </property>
        <property name="minValue" stdset="0">
         <double>0.000000000000000</double>
        </property>
        <property name="maxValue" stdset="0">
         <double>1000000000.000000000000000</double>
        </property>
        <property name="decimals" stdset="0">
         <number>10</number>
        </property>
       </widget>
      </item>
      <item row="0" column="3" rowspan="2">
       <widget class="QGroupBox" name="Scale_box">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>1</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
          <strikeout>false</strikeout>
          <kerning>true</kerning>
         </font>
        </property>
        <property name="title">
         <string/>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3" stretch="2,1">
         <property name="spacing">
          <number>1</number>
         </property>
         <property name="topMargin">
          <number>1</number>
         </property>
         <property name="bottomMargin">
          <number>1</number>
         </property>
         <item>
          <widget class="QRadioButton" name="lin_button">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
             <horstretch>1</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>24</height>
            </size>
           </property>
           <property name="text">
            <string>Linear</string>
           </property>
           <property name="iconSize">
            <size>
             <width>12</width>
             <height>12</height>
            </size>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
           <attribute name="buttonGroup">
            <string notr="true">scale_button_group</string>
           </attribute>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="log_button">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
             <horstretch>1</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>24</height>
            </size>
           </property>
           <property name="text">
            <string>Logarithmic</string>
           </property>
           <property name="iconSize">
            <size>
             <width>12</width>
             <height>12</height>
            </size>
           </property>
           <attribute name="buttonGroup">
            <string notr="true">scale_button_group</string>
           </attribute>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>floatLineEdit</class>
   <extends>QLineEdit</extends>
   <header>gui/resources/widgets/floatLineEdit.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>Var_dd</tabstop>
  <tabstop>from_entry</tabstop>
  <tabstop>to_entry</tabstop>
  <tabstop>lin_button</tabstop>
  <tabstop>log_button</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>Var_dd</sender>
   <signal>currentTextChanged(QString)</signal>
   <receiver>variable_from_to_scale</receiver>
   <slot>on_var_change(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>50</x>
     <y>66</y>
    </hint>
    <hint type="destinationlabel">
     <x>5</x>
     <y>59</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>scale_button_group</sender>
   <signal>buttonClicked(QAbstractButton*)</signal>
   <receiver>variable_from_to_scale</receiver>
   <slot>on_scale_change(QAbstractButton*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>257</x>
     <y>46</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>from_entry</sender>
   <signal>textChanged(QString)</signal>
   <receiver>variable_from_to_scale</receiver>
   <slot>on_from_change()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>205</x>
     <y>63</y>
    </hint>
    <hint type="destinationlabel">
     <x>509</x>
     <y>33</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>to_entry</sender>
   <signal>textChanged(QString)</signal>
   <receiver>variable_from_to_scale</receiver>
   <slot>on_to_change()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>322</x>
     <y>63</y>
    </hint>
    <hint type="destinationlabel">
     <x>509</x>
     <y>70</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>variable_changed(QString)</signal>
  <signal>from_changed(QString)</signal>
  <signal>to_changed(QString)</signal>
  <signal>scale_changed(QString)</signal>
  <slot>append_to_variable_list(QString)</slot>
  <slot>clear_variable_list()</slot>
  <slot>edit_from(QString)</slot>
  <slot>edit_to(QString)</slot>
  <slot>on_var_change(QString)</slot>
  <slot>on_from_change()</slot>
  <slot>on_to_change()</slot>
  <slot>on_scale_change(QAbstractButton*)</slot>
 </slots>
 <buttongroups>
  <buttongroup name="scale_button_group"/>
 </buttongroups>
</ui>
